docs/src/main/asciidoc/sending-emails.adoc

```diff
@@ -0,0 +1,205 @@
+
+== Creating the Maven Project
+
+Create a new project with the following command:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=sending-email-quickstart \
+    -Dextensions="mailer"
+----
+
+If you already have an existing project, add the `mailer` extension:
+
+[source]
+----
+mvn quarkus:add-extensions -Dextensions="mailer"
+----
+
+== Configuring the mailer
+
+The {project-name} mailer is using SMTP. In the `src/main/resources/application.properties` file, you need to configure the host, port, username, password as well as the other configuration aspect.
+Note that the password can also be configured using system properties and environment variables.
+
+Here is an example using _sendgrid_:
+
+[source]
+----
+quarkus.mailer.from=test@quarkus.io
+quarkus.mailer.host=smtp.sendgrid.net
+quarkus.mailer.port=465
+quarkus.mailer.ssl=true
+quarkus.mailer.username=....
+quarkus.mailer.password=....
+----
+
+The following table lists the properties that can be configured:
+
+[cols=*,options="header"]
+|===
+
+| Property | Type | Comment
+
+| from | String | the default _from_ address, without, all emails will need an explicit value
+| mock | Boolean | if set to `true`, the emails are not sent, just printed on the console.
+| bounceAddress | String | the default bounce address
+| host | String | the SMTP server host name
+| port | Integer | the SMTP server port
+| username | String | the username
+| password | String | the password
+| ssl | Boolean | Enables or disables SSL
+| trustAll | Boolean | Set whether to trust all certificates
+| maxPoolSize | Integer | Set the maximum pool size
+| ownHostName | String | The hostname to be used for `HELO/EHLO` and the `Message-ID`
+| keepAlive | Boolean | Set if connection pool is enabled, default is `true`
+| disableEsmtp | Boolean | Set if ESMTP should be tried as first command (EHLO).
+| startTLS | `NONE`, `OPTIONAL`, `REQUIRED` | Set the TLS security mode
+| login | `DISABLED`, `OPTIONAL`, `REQUIRED` | Set the login mode for the connection
+| authMethods | space-separated list | Set the allowed auth methods
+| keyStore | String | Path of the key store
+| keyStorePassword | String | Key store password
+|===
+
+== Sending simple emails
+
+In a JAX-RS resource, or in a bean, you can inject the mailer as follows:
+
+[source, java]
+----
+@Inject
+Mailer mailer;
+
+@Inject
+ReactiveMailer reactiveMailer;
+----
+
+There are 2 APIs:
+
+* `io.quarkus.mailer.Mailer` provides the imperative (blocking and synchronous) API;
+* `io.quarkus.mailer.ReactiveMailer` provides the reactive (non-blocking and asynchronous) API
+
+NOTE: `Mailer` is implemented on top of `ReactiveMailer`
+
+To send a simple email, proceed as follows:
+
+[source, java]
+----
+@GET
+@Path("/simple")
+public Response sendASimpleEmail() {
+    mailer.send(Mail.withText("to@acme.org", "A simple email from quarkus", "This is my body"));
```

Comment:
Can you split the snippet in two parts? Mail API usage and JAX-RS method impl?

Reply:
Done.